File: src/main/java/work/xeltica/craft/core/models/PlayerDataKey.java
package work.xeltica.craft.core.models;

/**
 * プレイヤーごとに保存するデータのキーを定義しています。
 * PlayerStore で使用します。
 * @author Xeltica
 */
public enum PlayerDataKey {
    CAT_MODE("cat"),
    NEWCOMER_TIME("newcomer_time"),
    BEDROCK_ACCEPT_DISCLAIMER("accept_disclaimer"),
    FIRST_SPAWN("first_spawn"),
    LAST_JOINED("last_joined"),
    GIVEN_PHONE("given_phone"),
    COUNTER_REGISTER_MODE("counter_register_mode"),
    COUNTER_REGISTER_NAME("counter_register_name"),
    COUNTER_REGISTER_IS_DAILY("counter_register_is_daily"),
    COUNTER_REGISTER_LOCATION("counter_register_location"),
    PLAYING_COUNTER_ID("counter_id"),
    PLAYING_COUNTER_TIMESTAMP("counter_time"),
    PLAYED_COUNTER("played_counter"),
    RECEIVED_LOGIN_BONUS("login_bonus"),
    RECEIVED_LOGIN_BONUS_SUMMER("login_bonus_summer"),
    BROKEN_BLOCKS_COUNT("broken_blocks_count"),
    MOB_DEX("mob_dex"),

    ;

    PlayerDataKey(String physicalKey) {
        this.physicalKey = physicalKey;
    }

    private final String physicalKey;

    public String getPhysicalKey() {
        return this.physicalKey;
    }
}

File: src/main/java/work/xeltica/craft/core/modules/player/PlayerDataKey.kt
package work.xeltica.craft.core.modules.player

/**
 * プレイヤーごとに保存するデータのキーを定義しています。
 * PlayerStore で使用します。
 * @author Xeltica
 */
object PlayerDataKey {
    val CAT_MODE = "cat"
    val NEWCOMER_TIME = "newcomer_time"
    val BEDROCK_ACCEPT_DISCLAIMER = "accept_disclaimer"
    val FIRST_SPAWN = "first_spawn"
    val LAST_JOINED = "last_joined"
    val GIVEN_PHONE = "given_phone"
    val COUNTER_REGISTER_MODE = "counter_register_mode"
    val COUNTER_REGISTER_NAME = "counter_register_name"
    val COUNTER_REGISTER_IS_DAILY = "counter_register_is_daily"
    val COUNTER_REGISTER_LOCATION = "counter_register_location"
    val PLAYING_COUNTER_ID = "counter_id"
    val PLAYING_COUNTER_TIMESTAMP = "counter_time"
    val PLAYED_COUNTER_COUNT = "counter_count"
    val RECEIVED_LOGIN_BONUS = "login_bonus"
    val RECEIVED_LOGIN_BONUS_SUMMER = "login_bonus_summer"
    val BROKEN_BLOCKS_COUNT = "broken_blocks_count"
    val MOB_DEX = "mob_dex"
}